alea/blueice_extended_model.py

```diff
@@ -0,0 +1,258 @@
+        """
+        super().__init__(parameter_definition=parameter_definition)
+        self._likelihood = self._build_ll_from_config(likelihood_config)
+        self.likelihood_names = [t["name"] for t in likelihood_config["likelihood_terms"]]
+        self.likelihood_names.append("ancillary_likelihood")
+        self.data_generators = self._build_data_generators()
+
+    # TODO analysis_space could be inferred from the data (assert that all sources have the same analysis space)
+
+    @classmethod
+    def from_config(cls, config_file_path: str) -> "BlueiceExtendedModel":
+        """Initializes the statistical model from a yaml config file.
+
+        Args:
+            config_file_path (str): Path to the yaml config file.
+
+        Returns:
+            BlueiceExtendedModel: Statistical model.
+        """
+        with open(config_file_path, "r") as f:
+            config = yaml.safe_load(f)
+        return cls(**config)
+
+    @property
+    def data(self) -> list:
+        """Returns the data of the statistical model."""
+        return super().data
+
+    @data.setter
+    def data(self, data: list):
+        """
+        Overrides default setter. Will also set the data of the blueice ll.
+        Data-sets are expected to be in the form of a list of one or more structured arrays-- representing the data-sets of one or more likelihood terms.
+        """
+        # iterate through all likelihood terms and set the science data in the blueice ll
+        # last entry in data are the generate_values
+        for d, ll_term in zip(data[:-1], self._likelihood.likelihood_list):
+            ll_term.set_data(d)
+
+        self._data = data
+
+    def get_expectation_values(self, **kwargs) -> dict:
+        """
+        Return total expectation values (summed over all likelihood terms with the same name)
+        given a number of named parameters (kwargs)
+        """
+        ret = dict()
+        for ll in self._likelihood.likelihood_list[:-1]:  # ancillary likelihood does not contribute
+            mus = ll(full_output=True, **kwargs)[1]
+            for n, mu in zip(ll.source_name_list, mus):
+                ret[n] = ret.get(n, 0) + mu
+        return ret
+
+    def _build_ll_from_config(self, likelihood_config: dict) -> "LogLikelihoodSum":
+        """
+        Iterate through all likelihood terms and build blueice ll
+
+        Args:
+            likelihood_config (dict): A dictionary defining the likelihood.
+        """
+        lls = []
+
+        # Iterate through each likelihood term in the configuration
+        for config in likelihood_config["likelihood_terms"]:
+            likelihood_object = locate(config["likelihood_type"])
+            if isinstance(likelihood_config["template_folder"], str):
+                template_folder_list = [likelihood_config["template_folder"]]
+            elif isinstance(likelihood_config["template_folder"], list):
+                template_folder_list = likelihood_config["template_folder"]
```

Comment:
So you mean always just add "alea/examples/templates/wimp_templates" to the end of the list? It probably doesn't hurt adding it, though I'm not sure how often this will be used (for the examples I think it is good to write things explicitly so that people know what to change when implementing their own model).